src/api/app/models/review.rb

```diff
@@ -21,7 +21,14 @@ class NotFoundError < APIError
   validates :reviewer, length: { maximum: 250 }
   validates :reason, length: { maximum: 65_534 }
 
-  validate :check_initial, on: [:create]
+  validates :user, presence: true, if: :by_user?, on: :create
```

Comment:
Should these validations be only on creation? I feel like I'm either missing something or `on: :create` should be removed.

Reply:
Hi Dani, yes, The PR was already like that, but yes, I think it shouldn't be just in the `:create` @coolo or @hennevogel what do you think? Should be possible to change the initial value anyway (i.e change a review from `by_project` to `by_user`) ?

Reply:
No, but it's possible to delete a user - and then review should stay intact. `:by_user` stays, `user` might be dead in old reviews.

Not so much of a problem with users (but technically a possiblity, think DSVGO :), but with project and package it happens *every day*

Reply:
@coolo ok, but then the following scenario would be possible when we do not limit the validation to `create`:

User create a review by project. Project is deleted. The review cannot be accepted, revoked, whatsoever, since the validation may fail.. right? So should we limit the validation just for creation?

Reply:
hm from comment looks like it should work:

```
 # The authoritative storage are the by_ attributes as even when a record (project, package ...) got deleted
  # the review should still be usable, however, the entity association is nullified
```

Reply:
What's the point of creating a new PR if you repeat the discussion from the old one? :)

https://github.com/openSUSE/open-build-service/pull/6282#discussion_r234075246

Reply:
But now I have the solution, based in your comment, without any BIAS and less noise.

if the valdiation `by_package` isn't limited to `on_create`, when you delete got deleted, you cannot change the state of the review anymore. I added some test cases to validate it. So looks like it make sense to have the validation just `on: :create` what do you think @coolo